### PR TITLE
refactor(misc): testngJava fix, systemrestli client, cache key fix, e…

### PR DESCRIPTION
**Pull request by david-leifker**
…lasticsearch index builder refactor

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by hsheth2 — APPROVED** [review]

File: metadata-io/src/main/java/com/linkedin/metadata/service/UpdateIndicesService.java [hunk 1]
```diff
@@ -90,25 +95,31 @@ public void setSearchDiffMode(boolean searchDiffMode) {
...
-    } else if (event.getChangeType() == ChangeType.DELETE) {
-      handleDeleteChangeEvent(event);
+    try {
+      if (UPDATE_CHANGE_TYPES.contains(event.getChangeType())) {
+        handleUpdateChangeEvent(event);
+      } else if (event.getChangeType() == ChangeType.DELETE) {
+        handleDeleteChangeEvent(event);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/service/UpdateIndicesService.java:122]: curious what was the IO exception we were looking at?

File: metadata-service/factories/src/main/java/com/linkedin/gms/factory/entity/update/indices/UpdateIndicesServiceFactory.java [hunk 2]
```diff
@@ -1,24 +1,44 @@
...
-    return new UpdateIndicesService(graphService, entitySearchService, timeseriesAspectService,
-        systemMetadataService, entityRegistry, searchDocumentTransformer);
+                                                   TimeseriesAspectService timeseriesAspectService,
+                                                   SystemMetadataService systemMetadataService,
+                                                   EntityRegistry entityRegistry, SearchDocumentTransformer searchDocumentTransformer,
+                                                   EntityIndexBuilders entityIndexBuilders) {
+    UpdateIndicesService updateIndicesService = new UpdateIndicesService(graphService, entitySearchService, timeseriesAspectService,
+            systemMetadataService, entityRegistry, searchDocumentTransformer, entityIndexBuilders);
+
+    if ("restli".equals(entityClientImpl)) {
```
Comment (jjoyce0510) [inline metadata-service/factories/src/main/java/com/linkedin/gms/factory/entity/update/indices/UpdateIndicesServiceFactory.java:38]: should this be a const?

File: metadata-service/services/src/main/java/com/linkedin/metadata/entity/EntityService.java [hunk 3]
```diff
@@ -297,4 +298,11 @@ Pair<Boolean, List<Pair<String, RecordTemplate>>> generateDefaultAspectsOnFirstW
    */
   @Nonnull
   BrowsePathsV2 buildDefaultBrowsePathV2(final @Nonnull Urn urn, boolean useContainerPaths) throws URISyntaxException;
+
+  /**
```
Comment (jjoyce0510) [inline metadata-service/services/src/main/java/com/linkedin/metadata/entity/EntityService.java:302]: This does not seem good. 

Entity Client should wrap entity service....
I don't quite understand why this is happening here

**Review by jjoyce0510 — COMMENTED** [review]
Had some comments

File: metadata-service/services/src/main/java/com/linkedin/metadata/entity/EntityService.java (hunk 3 above)
Comment (jjoyce0510) [reply metadata-service/services/src/main/java/com/linkedin/metadata/entity/EntityService.java:302]: I think this is just a lack of understanding for what's happening inside of entity service

**Pull request merged**